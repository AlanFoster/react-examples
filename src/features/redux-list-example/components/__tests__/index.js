/* eslint-env browser */
import React from 'react';
import renderer from 'react-test-renderer';
import ReduxListExample from '../';

jest.mock('react-virtualized/dist/commonjs/AutoSizer', function() {
  return function(props) {
    const renderCallback = props.children;

    return renderCallback({
      width: 600
    });
  };
});

it('renders the given repositories', () => {
  const repositories = [
    {
      id: 1,
      fullName: 'full-name-1',
      description: 'description-1',
      htmlUrl: 'html-url-1'
    },
    {
      id: 2,
      fullName: 'full-name-2',
      description: 'description-2',
      htmlUrl: 'html-url-2'
    }
  ];

  const getData = jest.fn();

  const component = renderer.create(
    <ReduxListExample repositories={repositories} getData={getData} />
  );

  expect(component).toMatchSnapshot();
  expect(getData).toHaveBeenCalled();
});
