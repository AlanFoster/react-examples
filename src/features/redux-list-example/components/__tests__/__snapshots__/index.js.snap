// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the given repositories 1`] = `
<div>
  <h2>
    Top Repositories
  </h2>
  <div
    aria-label="grid"
    className="ReactVirtualized__Grid ReactVirtualized__List repositories"
    id={undefined}
    onScroll={[Function]}
    role="grid"
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "boxSizing": "border-box",
        "direction": "ltr",
        "height": 300,
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
    tabIndex={0}
  >
    <div
      className="ReactVirtualized__Grid__innerScrollContainer"
      style={
        Object {
          "height": 140,
          "maxHeight": 140,
          "maxWidth": 600,
          "overflow": "hidden",
          "pointerEvents": "",
          "position": "relative",
          "width": "auto",
        }
      }
    >
      <div
        className="repositories__row"
        style={
          Object {
            "height": 70,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <div>
          <div>
            full-name-1
          </div>
          <div>
            description-1
          </div>
        </div>
      </div>
      <div
        className="repositories__row"
        style={
          Object {
            "height": 70,
            "left": 0,
            "position": "absolute",
            "top": 70,
            "width": "100%",
          }
        }
      >
        <div>
          <div>
            full-name-2
          </div>
          <div>
            description-2
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
